### ISSUE TITLE
terminated by signal SIGSEGV (Address boundary error)

### ISSUE THREAD

**Issue by dpc** [fish-shell/fish-shell#5409]
```
[I] 12-12 22:12 dpc@futex ~ [139]> fish --version
fish, version 2.7.1
[I] 12-12 22:12 dpc@futex ~> echo $version
2.7.1
[I] 12-12 22:12 dpc@futex ~> uname -a
Linux futex.dpc.pw 4.14.79 #1-NixOS SMP Sun Nov 4 13:52:51 UTC 2018 x86_64 GNU/Linux
[I] 12-12 22:12 dpc@futex ~> echo $TERM
xterm-256color
[I] 12-12 22:12 dpc@futex ~> sh -c 'env HOME=$(mktemp -d) fish -C "set -gx fish_history \'\'"'
fish: “sh -c 'env HOME=$(mktemp -d) fi…” terminated by signal SIGSEGV (Address boundary error)
```

**Comment by faho** [comment]
Duplicate of #4619

**Comment by faho** [comment]
You will find this has been fixed in the freshly released [3.0 beta](https://github.com/fish-shell/fish-shell/releases/tag/3.0b1). Get it while it's still hot! 3.0!
